docs/resources/pumls/pipeline.puml

```diff
@@ -0,0 +1,25 @@
+@startuml Pipeline
+
+!includeurl ./style.puml
+
+package "Pipeline" {
+
+class Pipeline {
```

Comment:
Add `__init__(provider: StreamProvider, infer_engine_info: InferenceInfo)` and `__iter__(): Iterator[Tuple[str, Dict[str, Any]]]` functions